src/NHibernate/Proxy/ProxyBuilderHelper.cs

```diff
@@ -0,0 +1,188 @@
+#region Credits
+
+// This work is based on LinFu.DynamicProxy framework (c) Philip Laureano who has donated it to NHibernate project.
+// The license is the same of NHibernate license (LGPL Version 2.1, February 1999).
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace NHibernate.Proxy
+{
+	internal static class ProxyBuilderHelper
+	{
+#if NETFX
+		private static bool _saveAssembly;
+		private static string _saveAssemblyPath;
+
+		// Called by reflection
+		internal static void EnableDynamicAssemblySaving(bool enable, string saveAssemblyPath)
+		{
+			_saveAssembly = enable;
+			_saveAssemblyPath = saveAssemblyPath;
```

Comment:
`DefaultProxyAssemblyBuilder` was enabling that with compilation directives, which was not very practical. Moreover, it was not handling the save path, which was further complicating using the feature with potentially more file system access right troubles.